src/data/util/Optional.java

```diff
@@ -0,0 +1,62 @@
+package data.util;
+
+import com.sun.javafx.beans.annotations.NonNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.NoSuchElementException;
+
+public class Optional<T> {
```

Comment:
consider javadoc but.... bleh. I know what the `Optional` is. heck, everyone should know **exactly** what the `Optional` is :D